### ISSUE TITLE
ACR authentication

### ISSUE THREAD

**Issue by romanzdk** [GoogleContainerTools/kaniko#2059]
I am trying to push my image into Azure Container Registry. I am creating config.json like this:
```
cat << EOF > /kaniko/.docker/config.json
{
  "auths": {
        "$(ACR)": {
          "username":"$(username)",
          "password":"$(password)"
        }
  }
}
EOF
```

However I am getting `error checking push permissions -- make sure you entered the correct tag name, and that you are authenticated correctly, and try again: checking push permission for "***/example-project:latest": creating push check transport for *** failed: Get "[https://***/v2/"](https://%2A%2A%2A/v2/%22): EOF`

**Comment by romanzdk** [comment]
No one?

**Comment by Chuvisco88** [comment]
Getting the same error :/

**Comment by romanzdk** [comment]
No, we switched to completely different solution, nonetheless, it looks like this approach is simply not supported by Kaniko.

**Comment by dza89** [comment]
You have to set three environment variables:
`AZURE_CLIENT_ID`
`AZURE_CLIENT_SECRET`
`AZURE_TENANT_ID`

And do this:
`mkdir -p /kaniko/.docker`
`echo "{\"credHelpers\":{\"cptazprod.azurecr.io\":\"acr-env\"}}" > /kaniko/.docker/config.json`

**Comment by nunopedrosilva** [comment]
This is an old post, but anyways. I was just wondering, whether it is possible to do this without a service principal, but instead using a managed identity?
Thanks!

**Comment by twanbeeren** [comment]
Still no update on this? It would be really helpful

**Comment by DeanHnter** [comment]
Im having this same issue now also, previously it worked but now using latest image it seems not to accept base64 username:password for the auth and I get unauthorized warnings.

**Comment by fehemilec** [comment]
@DeanHnter how did you use the service principal? I can't make it work. Could you please share your solution?
